text/0000-ancillary-package-module-standardization.md

```diff
@@ -0,0 +1,240 @@
+the global `jQuery`, with the `getJSON` method added on to it.
+
+## File System Layout
+
+Packages will adopt the following layout:
+
+* `src` - Source ES2015+ or TypeScript. Layout inside `src` is up to
+  authors.
+* `dist` - Per Broccoli defaults, built output is placed in the `dist`
+  directory. The `package.json`, tests, and other configuration
+  information should consume the files in `dist`.
+  * `js` - JavaScript standard/ES2015/ES6 module syntax build
+  * `cjs` - CommonJS build
+  * `umd/package-name.js` - UMD build
+
+## Build Pipeline
+
+Packages will standardize on using Ember CLI as their build tool. This
+maximizes familiarity for Ember developers becoming contributors to the
+framework.
+
+We will provide a package on npm, `broccoli-module-alchemist`, that
+encapsulates these concerns. Using it in a package is as simple as:
+
+1. `npm install ember-cli --save-dev`
+2. `npm install broccoli-module-alchemist --save-dev`
+3. Require `broccoli-module-alchemist` from `ember-cli-build.js` and
+   return it as the Broccoli tree.
+4. Run `ember build`
+
+Here's an example build file:
+
+```js
+// ember-cli-build.js
+
+var alchemist = require('broccoli-module-alchemist');
+
+module.exports = function() {
+  return alchemist();
+}
+```
+
+### Under the Hood
+
+We use two tools to produce the desired output formats:
+
+1. The TypeScript compiler, which converts TypeScript and ES2015+ to
+   ES5, in both CommonJS and standard module syntax.
+2. Rollup, to package the ES5 source into legacy bundle formats like
+   UMD.
+
+## Examples
+
+| Repo                 | Description                                                                                               | PR                                                 |
+|----------------------|-----------------------------------------------------------------------------------------------------------|----------------------------------------------------|
+| [fastboot](https://github.com/ember-fastboot/fastboot) (Node-only) | Uses broccoli-module-alchemist to utilize Node 4's ES2015 syntax features while remaining 0.12 compatible | https://github.com/ember-fastboot/fastboot/pull/74 |
+
+# How We Teach This
+
+Because this is an internal infrastructure concern, it does not have
+user-facing impact. That said, we should ensure that ease-of-use and
+maintainability are priorities to enable new contributors to get
+started.
+
+Because users who need this information are intermediate to advanced,
+this RFC itself should serve as a good starting point. Additionally,
+configuration information and how-tos will be incorporated into
+`broccoli-module-alchemist`'s README.
+
+# Drawbacks
+
+The biggest drawback is that this change may introduce temporary
+instability into the ecosystem. Many packages may rely on non-standard
+semantics or syntax, and updating them may divert work that could go
+elsewhere.
+
+Additionally, this RFC does not go so far as attempting to standardize
```

Comment:
@tomdale I think we should use ember-cli, and continue the effort to reduce "weight" of it, as that will help our whole ecosystem. For example, making bower an optional dependency will cut the filesize and filecount by about 30%.

My motivation for this is as follows, rather then re-inventing the bits that are good in ember-cli (there is a very large number of these things) outside of it, we should take that energy in removing the heaviness of ember-cli's default install. Fixing the heaviness, will also provide us larger leverage. 

On this note, the effort is actually on-going @kellyselden is working on it, I am as well, as are others.

So in the interim, lets take incur this cost.